packages/inter-protocol/src/proposals/replace-scaledPriceAuthorities.js

```diff
@@ -0,0 +1,216 @@
+import { makeTracer, deeplyFulfilledObject } from '@agoric/internal';
+import { E } from '@endo/far';
+import { makeRatio } from '@agoric/zoe/src/contractSupport/index.js';
+import { parseRatio } from '@agoric/zoe/src/contractSupport/ratio.js';
+
+import { reserveThenGetNames, scaledPriceFeedName } from './utils.js';
+
+const trace = makeTracer('replaceScaledPA', true);
+
+/**
+ * Copied with minor modification from addAssetToVault.js because a previous
```

Comment:
please factor out the common code so they stay in sync as we evolve them. also makes code review much faster to see it as code movement than apparently new lines

Reply:
done